sdk/identity/azure-identity/azure/identity/aio/_credentials/chained.py

```diff
@@ -31,15 +32,21 @@ class ChainedTokenCredential(SyncChainedTokenCredential):
         .. note:: This method is called by Azure SDK clients. It isn't intended for use in application code.
 
         :param str scopes: desired scopes for the token
-        :raises ~azure.core.exceptions.ClientAuthenticationError:
+        :raises ~azure.core.exceptions.ClientAuthenticationError: no credential in the chain provided a token
         """
         history = []
         for credential in self.credentials:
             try:
                 return await credential.get_token(*scopes, **kwargs)
-            except ClientAuthenticationError as ex:
+            except CredentialUnavailableError as ex:
+                # credential didn't attempt authentication because it lacks required data or state -> continue
                 history.append((credential, ex.message))
             except Exception as ex:  # pylint: disable=broad-except
+                # credential failed to authenticate, or something unexpectedly raised -> raise
                 history.append((credential, str(ex)))
+                error_message = self._get_error_message(history)
+                raise ClientAuthenticationError(message=error_message)
```

Comment:
You already have line L51 & L52. Why you need it again here?

Reply:
Sure, this could be a break instead, I'll change it.